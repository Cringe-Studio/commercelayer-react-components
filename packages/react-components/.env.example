# Unit tests
VITE_TEST_CLIENT_ID=
VITE_TEST_ENDPOINT=
VITE_TEST_MARKET_ID=
VITE_TEST_USERNAME=
VITE_TEST_PASSWORD=
VITE_TEST_CLIENT_ID_INTEGRATION=
VITE_TEST_CLIENT_SECRET=
