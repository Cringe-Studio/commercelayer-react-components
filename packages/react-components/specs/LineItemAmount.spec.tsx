import React from 'react'
import { LineItemAmount } from '../packages/react-components/src'
import renderer from 'react-test-renderer'
import components from '../#config/components'
import Parent from '../#components/utils/Parent'

const propTypes = components.LineItemAmount.propTypes

test('<LineItemAmount/>', () => {
  expect.assertions(4)
  const component = renderer.create(<LineItemAmount />)
  const tree = component.toJSON()
  const root = component.toTree()
  const proptypes = root.type['propTypes']
  expect(tree).toMatchSnapshot()
  expect(proptypes.children).toBe(propTypes.children)
  expect(proptypes.format).toBe(propTypes.format)
  expect(proptypes.type).toBe(propTypes.type)
})

test('<LineItemAmount check children format />', () => {
  expect.assertions(4)
  const component = renderer.create(
    <LineItemAmount format="float" type="unit" />
  )
  const tree = component.toJSON()
  const root = component.toTree()
  const rendered = root.rendered
  expect(tree).toMatchSnapshot()
  expect(rendered.type).toBe('span')
  expect(root.props.format).toBe('float')
  expect(root.props.type).toBe('unit')
})

test('<LineItemAmount with custom children />', () => {
  expect.assertions(8)
  const CustomComponent = (props) => <span>{props.label}</span>
  const component = renderer.create(
    <LineItemAmount>{CustomComponent}</LineItemAmount>
  )
  const tree = component.toJSON()
  const root = component.toTree()
  const rendered = root.rendered
  const childRendered = root.rendered.rendered
  expect(tree).toMatchSnapshot()
  expect(rendered.type).toBe(Parent)
  expect(rendered.props.price).toBe('')
  expect(childRendered.nodeType).toBe('component')
  expect(childRendered.type).toBe(CustomComponent)
  expect(childRendered.props.price).toBe('')
  expect(childRendered.rendered.nodeType).toBe('host')
  expect(childRendered.rendered.type).toBe('span')
})
