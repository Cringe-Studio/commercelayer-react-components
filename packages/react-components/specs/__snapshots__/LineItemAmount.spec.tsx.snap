// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LineItemAmount check children format /> 1`] = `
<span>
  
</span>
`;

exports[`<LineItemAmount with custom children /> 1`] = `<span />`;

exports[`<LineItemAmount/> 1`] = `
<span>
  
</span>
`;
